block/manager.go

```diff
@@ -189,19 +205,51 @@ func (m *Manager) AggregationLoop(ctx context.Context) {
 		time.Sleep(delay)
 	}
 
+	//var timer *time.Timer
 	timer := time.NewTimer(0)
 
-	for {
-		select {
-		case <-ctx.Done():
-			return
-		case <-timer.C:
-			start := time.Now()
-			err := m.publishBlock(ctx)
-			if err != nil {
-				m.logger.Error("error while publishing block", "error", err)
+	if !lazy {
+		for {
+			select {
+			case <-ctx.Done():
+				return
+			case <-timer.C:
+				start := time.Now()
+				err := m.publishBlock(ctx)
+				if err != nil {
+					m.logger.Error("error while publishing block", "error", err)
+				}
+				timer.Reset(m.getRemainingSleep(start))
+			}
+		}
+	} else {
+		<-timer.C
+		for {
+			select {
+			case <-ctx.Done():
+				return
+			// the buildBlock channel is signalled when Txns become available
+			// in the mempool, or after transactions remain in the mempool after
+			// building a block.
+			case <-m.txsAvailable:
+				m.logger.Debug("Lazy mode: txs available! Starting block building...")
+				if !m.buildingBlock {
+					m.buildingBlock = true
+					timer.Reset(1 * time.Second)
+				}
+			case <-timer.C:
+				// build a block with all the transactions received in the last 1 second
+				m.logger.Debug("Block building time elapsed.")
```

Comment:
same here. you can remove this. generally prefer debug if you printing parameter values.